### PR TITLE
Add pre-push Git Hook and addPrePushGitHook task

### PR DESCRIPTION
**Pull request by M0rgho**
This hook runs spotlessApply before pushing to remote branch.

Run ./gradlew addPreCommitGitHookOnBuild to enable this hook.

### CODE REVIEW

**Comment by M0rgho** [comment]
To merge this pull requests, two reviews are needed one from linux and one from windows.

File: .scripts/pre-push [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+# Run spotlessApply using ./gradlew before pushing
+
+COMMAND="./gradlew --quiet spotlessCheck"
```
Comment (DavidJones1102) [inline .scripts/pre-push:5]: will this only check if project is well formatted or it makes autoformatting also?

**Review by DavidJones1102 — APPROVED** [review]

**Comment by funsafe-math** [comment]
It may be a good idea to add setup commands to README.md, otherwise they will quickly become lost.
IE: " ./gradlew addPreCommitGitHookOnBuild " > README.md

File: .scripts/pre-push (hunk 1 above)
Comment (M0rgho) [reply .scripts/pre-push:5]: only checks

**Pull request merged**